### PR TITLE
simple change to reduce memory usage

### PR DESCRIPTION
**Pull request by fabianmurariu**
### What changes were proposed in this pull request?
small improvement to memory usage, about 20%

### Why are the changes needed?
we're memory hungry

### Does this PR introduce any user-facing change? If yes is this documented?
No

### How was this patch tested?
unit tests

### Issues

#1667 

### Are there any further changes required?
yes

### CODE REVIEW

File: raphtory/src/core/entities/edges/edge_store.rs [hunk 1]
```diff
@@ -26,17 +26,39 @@ use ouroboros::self_referencing;
...
+    fn deref(&self) -> &Self::Target {
+        &self.0.layer
+    }
+}
+
+impl DerefMut for EdgeLayerMut<'_> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0.layer
+    }
 }
```
Comment (ljeub-pometry) [inline raphtory/src/core/entities/edges/edge_store.rs:62]: Doesn't seem useful to make a wrapper?

File: raphtory/src/core/entities/edges/edge_store.rs [hunk 2]
```diff
@@ -95,32 +117,27 @@ impl EdgeStore {
...
-        if self.layers.len() <= layer_id {
-            self.layers.resize_with(layer_id + 1, Default::default);
+        self.data.len()
+    }
+    fn get_or_allocate_layer(&mut self, layer_id: usize) -> EdgeLayerMut {
+        if self.data.len() <= layer_id {
+            self.data.resize_with(layer_id + 1, Default::default);
         }
-        &mut self.layers[layer_id]
+        EdgeLayerMut(&mut self.data[layer_id])
```
Comment (ljeub-pometry) [inline raphtory/src/core/entities/edges/edge_store.rs:126]: could just return `&mut self.data[layer_id].layer` here?

**Review by ljeub-pometry — APPROVED** [review]
Not sure why you added an extra wrapper type but looks good

File: raphtory/src/core/entities/edges/edge_store.rs (hunk 2 above)
Comment (fabianmurariu) [reply raphtory/src/core/entities/edges/edge_store.rs]: got carried away, it's gone now

File: raphtory/src/core/entities/edges/edge_store.rs (hunk 1 above)
Comment (fabianmurariu) [reply raphtory/src/core/entities/edges/edge_store.rs]: gone

**Pull request merged**